### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Math-ias**
This should have been in master a looong time ago.

So let's merge it now to follow the philosophy of having untested but checked code in the master branch and tags for when the system is stable and working.

Good luck :smile:

### CODE REVIEW

**Comment by jakekinsella** [comment]
Just a heads up, I will be waiting to merge this one because I have some stuff to talk about with @Math-ias and @simon-andrews in person (This could be done tomorrow morning).

**Comment by jakekinsella** [comment]
Another thing I will say about this is that there is a lot to be done to this code in terms of code standards, @jakebacker :)

**Comment by Math-ias** [comment]
Do you think we ought to, as a response to this, have a bunch of tabs to spaces git hooks?

File: src/org/robockets/tshirtcannon/subsystems/cannon/AdjustCannon.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Move the cannon up or down
+ */
+public class AdjustCannon extends Command {
+    ZAxisRelativeDirection dir;
+    
+    public AdjustCannon(ZAxisRelativeDirection dir) {
+        requires(Robot.cannonSubsystem);
+        this.dir = dir;
+    }
```
Comment (jakekinsella) [inline src/org/robockets/tshirtcannon/subsystems/cannon/AdjustCannon.java:16]: Newline after this

File: src/org/robockets/tshirtcannon/subsystems/cannon/Cannon.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    /**
+     * Rotate the base of the cannon.
+     * @param dir Direction to rotate in. Must be left or right.
+     * @deprecated Deprecated until the hardware exists for this
+     */
+	@Deprecated
+    public void rotateBase(XAxisRelativeDirection dir) {
+    	if (dir == XAxisRelativeDirection.LEFT) {
+    		//TODO: spin left
+    	} else if(dir == XAxisRelativeDirection.RIGHT) {
```
Comment (jakekinsella) [inline src/org/robockets/tshirtcannon/subsystems/cannon/Cannon.java]: Space after if

File: src/org/robockets/tshirtcannon/subsystems/cannon/Cannon.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    /**
+     * Adjust the angle of the cannon.
+     * @param dir Direction to move the cannon. Must be up or down.
+     */
+    public void adjustAngle(ZAxisRelativeDirection dir) {
+        if (dir == ZAxisRelativeDirection.UP) {
+            //if (RobotMap.cannonZAxisEncoder.get() <= MAX_UP_DOWN_DISTANCE) {
+                RobotMap.cannonXAxisMotorController.set(ADJUST_SPEED);
+            //}
+        } else if(dir == ZAxisRelativeDirection.DOWN) {
```
Comment (jakekinsella) [inline src/org/robockets/tshirtcannon/subsystems/cannon/Cannon.java]: Space after if

File: src/org/robockets/tshirtcannon/subsystems/cannon/FireCannon.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.robockets.tshirtcannon.subsystems.cannon;
+
+import edu.wpi.first.wpilibj.command.CommandGroup;
+
+/**
+ * Fire one tshirt.
+ */
+public class FireCannon extends CommandGroup {
+    
+    public  FireCannon() {
```
Comment (jakekinsella) [inline src/org/robockets/tshirtcannon/subsystems/cannon/FireCannon.java:10]: Is there a double space?

File: src/org/robockets/tshirtcannon/subsystems/DriveTrain.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package org.robockets.tshirtcannon.subsystems;
+
+import org.robockets.tshirtcannon.RobotMap;
+import org.robockets.tshirtcannon.commands.Drive;
+
+import edu.wpi.first.wpilibj.RobotDrive;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ *
```
Comment (jakekinsella) [inline src/org/robockets/tshirtcannon/subsystems/DriveTrain.java]: Javadocs?

**Comment by Math-ias** [comment]
Fixed it all :smile:

**Comment by Math-ias** [comment]
Jake said :+1: Woopdeedoo

**Comment by simon-andrews** [comment]
What branch do we develop on now? Testing? Cannon? Master?

**Comment by Math-ias** [comment]
We develop now on branches, these are reviewed for quality, and they are submitted to master for testing.

**Pull request merged**